src/ol/reproj.js

```diff
@@ -247,42 +247,44 @@ export function render(
     extend(sourceDataExtent, src.extent);
   });
 
-  const canvasWidthInUnits = getWidth(sourceDataExtent);
-  const canvasHeightInUnits = getHeight(sourceDataExtent);
-  const stitchContext = createCanvasContext2D(
-    Math.round((pixelRatio * canvasWidthInUnits) / sourceResolution),
-    Math.round((pixelRatio * canvasHeightInUnits) / sourceResolution),
-    canvasPool
-  );
-
-  if (!interpolate) {
-    stitchContext.imageSmoothingEnabled = false;
-  }
-
-  const stitchScale = pixelRatio / sourceResolution;
+  let stitchContext;
+  if (sourceResolution !== 0 || sources.length !== 1 || gutter !== 0) {
```

Comment:
Instad of introducing a special meaning of `0`, wouldn't it be more a more general solution to compare `sourceResolution` and `targetResolution`, and enter this code path when they are equal?

Reply:
Equal or not we always want to do it for Image reprojection (unless Firefox because it is too slow), and do not want to do it for Tile reprojection (we could if there was only one source tile, but the quality would change as soon as a second tile comes into view).  Using zero avoids passing another parameter.

Reply:
I see the rationale behind it, I'm just not a fan of misusing existing properties with special values to trigger entering different code  paths. A clean solution would be to use the same sourceResolution and targetResolution, and add a new boolean property `stitch` or something.